modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/DynamicConfigExpirationTestCase.java

```diff
@@ -85,6 +129,38 @@ public void expirationWorksAfterRestartingSource() throws Exception {
     doNotExpireDynamicConfigWithCustomExpirationUsedBySource();
   }
 
+  @Test
```

Comment:
these test are not wrong, don't delete them, but they are only helpful for troubleshooting. These test are testing the implementation, not the behavior. This is missing tests that verify that a config which has exceeded its TTL, it's never expired because it still has an open stream. Afterwards, the same test needs to close the stream and verify that the config is indeed expired. Test coverage here is not enough

Reply:
Yes I 100% agree. Until https://www.mulesoft.org/jira/browse/MULE-18774 is solved, configs are not expired until the flow finishes, so the only option we have is to check the counter (thus checking implementation rather than behavior). Actually all the dataweave waits are not needed for these tests, they are just there for when the other issues get solved. But I don't think this is too clear, so I will split the tests in 2: 1 group checks behavior and the other group checks implementation with a comment that says to remove when MULE-18774 is fixed.